### PR TITLE
prov/util: fix uninitialized variable

### PR DESCRIPTION
**Pull request by aingerson**
Set iface and device when calling util regattr

Signed-off-by: aingerson <alexia.ingerson@intel.com>

### CODE REVIEW

**Comment by shefty** [comment]
It's not fixing a coverity defect. :) It's fixing an uninitialized variable.  The problem was only reported by coverity.

**Comment by aingerson** [comment]
> It's not fixing a coverity defect. :) It's fixing an uninitialized variable. The problem was only reported by coverity.

I don't know what you're talking about. 😬

File: prov/util/src/util_mr_map.c [hunk 1]
```diff
@@ -280,6 +280,9 @@ int ofi_mr_regv(struct fid *fid, const struct iovec *iov,
 	attr.offset = offset;
 	attr.requested_key = requested_key;
 	attr.context = context;
+	attr.iface = FI_HMEM_SYSTEM;
+	attr.device.reserved = 0;
+
 	return ofi_mr_regattr(fid, &attr, flags, mr_fid);
```
Comment (shefty) [inline prov/util/src/util_mr_map.c:286]: This is exposing a backward compatibility problem.  ofi_mr_regattr() ends up being exposed directly to applications by some providers.  Unless the app has updated to the latest API version, they will not have those fields defined.  They might not even exist for compiled apps.

ofi_mr_regattr() needs to be updated to check the application version specified with opening the domain.  Unless the version indicates that those fields are present and the app has set them, we cannot touch them.  It looks like the problem needs to be handled in dup_mr_attr.

Comment (shefty) [reply prov/util/src/util_mr_map.c:286]: @rwespetal - doing a search on the code, I think EFA may have the same issue.  In efa_mr.c::efa_mr_reg_impl() the code accesses the iface field of the fi_mr_attr without first verifying that the application has been built with that field present and initialized (indicated by the app requested FI_HMEM capability).

Comment (rwespetal) [reply prov/util/src/util_mr_map.c:286]: I see, thanks for the heads up.

Comment (shefty) [reply prov/util/src/util_mr_map.c:286]: @aingerson - I'm preparing a fix to ofi_mr_regattr, and the socket provider.
@rwespetal - My intent is to allow ofi_mr_map_insert() to assume that that latest version of the structure is available, just so I don't have to pass through the version and capability bits.
It turns out the MR compatibility problem dates back to the 1.5 release and the addition of auth_keys... :/

Comment (aingerson) [reply prov/util/src/util_mr_map.c:286]: Can you check mr_map->prov->fi_version in ofi_mr_map_insert?

Comment (shefty) [reply prov/util/src/util_mr_map.c:286]: That doesn't work.  You need to version passed in by the app when it opens the fabric.  Plus the man pages say that for HMEM, the fields are ignored unless FI_HMEM has been specified.  So, the version is needed to check the auth_key fields, an the FI_HMEM cap bit set on the domain for the iface fields.  Both of those would need to be passed in.

But ofi_mr_map_insert() isn't really the best place to check for compatibility.  Long term it's easier to catch it at the call-in point, and let the code underneath not have to deal with it.

**Pull request merged**